### PR TITLE
Support multiple artist and other tags in FLAC/OGG files

### PR DESCRIPTION
**Pull request by OzGav**
Added support for multiple Vorbis comment fields in FLAC/OGG files per the official Vorbis specification, where multiple ARTIST or ALBUMARTIST fields can be used instead of a single ARTISTS tag with delimiters. This allows users who tag their files according to the Vorbis spec (e.g., ARTIST=Artist 1, ARTIST=Artist 2, ARTIST=Artist 3 as separate fields) to have their multi-artist tracks properly recognized.

Also added a dedicated _parse_vorbis_tags() function for FLAC/OGG files with feature parity to the MP4 parser (date, lyrics, compilation, ReplayGain, MusicBrainz album type), and created an integration test with a real FLAC test file.

Satisfies https://github.com/orgs/music-assistant/discussions/4595

### CODE REVIEW

File: music_assistant/providers/filesystem_local/constants.py [hunk 1]
```diff
@@ -137,6 +137,7 @@
     "ape",
     "mpc",
     "mp2",
+    "m2a",
     "mp1",
```
[the remark this answers is not in the document]
Comment (OzGav) [reply music_assistant/providers/filesystem_local/constants.py:141]: Avoiding scope creep is important but for six characters I felt the overhead of an additional PR was unnecessary

File: music_assistant/helpers/tags.py [hunk 2]
```diff
@@ -925,10 +926,149 @@ def _parse_id3_tags(tags: dict[str, Any]) -> dict[str, Any]:
...
+            # Multiple ALBUMARTIST fields - treat as authoritative list
+            result["albumartists"] = albumartist_values
+        else:
+            result["albumartist"] = albumartist_values[0]
+
+    # Explicit ARTISTS tag takes precedence if present
+    if artists := _vorbis_get_multi(tags, "ARTISTS"):
+        result["artists"] = artists
+
+
```
[the remark this answers is not in the document]
Comment (OzGav) [reply music_assistant/helpers/tags.py:982]: There is no ALBUMARTISTS (plural) tag in the Vorbis spec. Docstring fixed

File: music_assistant/helpers/tags.py [hunk 3]
```diff
@@ -16,6 +16,7 @@
 import mutagen
 from music_assistant_models.enums import AlbumType
 from music_assistant_models.errors import InvalidDataError
+from mutagen._vorbis import VCommentDict
 from mutagen.mp4 import MP4Tags
```
[the remark this answers is not in the document]
Comment (OzGav) [reply music_assistant/helpers/tags.py:20]: Technically correct but I think low risk. VCommentDict is the base class for all vorbis-style tags in mutagen and has been stable for years.

File: tests/core/test_tags.py [hunk 4]
```diff
@@ -1,15 +1,17 @@
 """Tests for parsing audio file tags (ID3, MP4/AAC, etc.)."""
 
 import pathlib
+from unittest.mock import MagicMock
 
 from music_assistant.constants import UNKNOWN_ARTIST
 from music_assistant.helpers import tags
-from music_assistant.helpers.tags import split_artists
+from music_assistant.helpers.tags import _parse_vorbis_tags, split_artists
```
[the remark this answers is not in the document]
Comment (OzGav) [reply tests/core/test_tags.py:8]: I’m not super familiar with this but checked with Claude and its response seems valid

This review comment is not particularly valid in this context. Here’s why:
	1.	The tests that import _parse_vorbis_tags are unit tests with mocked VCommentDict objects — they’re specifically testing the parsing logic in isolation, which is exactly what unit tests should do. Testing through parse_tags_mutagen or async_parse_tags would require real audio files for every edge case (multiple artist fields, precedence of ARTISTS over ARTIST, etc.), making the tests slower, more brittle, and harder to maintain.
	2.	The file already tests through the public API too — test_parse_metadata_from_flac_with_multiple_artist_fields uses async_parse_tags with a real FLAC fixture. The mock-based tests complement this by covering specific edge cases that would be tedious to create fixture files for.
	3.	The _ prefix convention in Python is a hint, not a contract. Within the same project’s test suite, testing private helpers directly is completely normal and expected. This isn’t a third-party consumer — it’s the project’s own tests.
	4.	Making _parse_vorbis_tags public just to satisfy the test import would be the wrong response — it’s genuinely an internal helper that shouldn’t be part of the module’s public API.

**Review by marcelveldt — APPROVED** [review]
Nice work @OzGav !
